arch/arc/core/cache.c

```diff
@@ -55,27 +74,294 @@ static void dcache_dc_ctrl(uint32_t dcache_en_mask)
 	}
 }
 
+static bool pae_exists(void)
+{
+	uint32_t bcr = z_arc_v2_aux_reg_read(_ARC_V2_MMU_BUILD);
+
+	return 1 == FIELD_GET(MMU_BUILD_PAE_MASK, bcr);
+}
+
+#if defined(CONFIG_SLC)
+static void slc_enable(void)
+{
+	uint32_t val = z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL);
+
+	val &= ~SLC_CTRL_DISABLE_MASK;
+	z_arc_v2_aux_reg_write(_ARC_V2_SLC_CTRL, val);
+}
+
+static void slc_high_addr_init(void)
+{
+	z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_END1, 0);
+	z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_START1, 0);
+}
+
+static void slc_flush_region(void *start_addr_ptr, size_t size)
+{
+	uintptr_t start_addr = (uintptr_t)start_addr_ptr;
+	uintptr_t end_addr;
+	uint32_t ctrl;
+
+	K_SPINLOCK(&slc_lock) {
+		ctrl = z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL);
+
+		ctrl &= ~SLC_CTRL_RGN_OP_MASK;
+
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_CTRL, ctrl);
+
+		/*
+		 * Lower bits are ignored, no need to clip
+		 * END needs to be setup before START (latter triggers the operation)
+		 * END can't be same as START, so add (l2_line_sz - 1) to sz
+		 */
+		end_addr = start_addr + size + CONFIG_SLC_LINE_SIZE - 1;
+
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_END, end_addr);
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_START, start_addr);
+
+		/* Make sure "busy" bit reports correct status, see STAR 9001165532 */
+		z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL);
+		while (z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL) & SLC_CTRL_BS_MASK) {
+			/* Do Nothing */
+		}
+	}
+}
+
+static void slc_invalidate_region(void *start_addr_ptr, size_t size)
+{
+	uintptr_t start_addr = (uintptr_t)start_addr_ptr;
+	uintptr_t end_addr;
+	uint32_t ctrl;
+
+	K_SPINLOCK(&slc_lock) {
+		ctrl = z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL);
+
+		ctrl &= ~SLC_CTRL_IM_MASK;
+
+		ctrl &= ~SLC_CTRL_RGN_OP_MASK;
+		ctrl |= FIELD_PREP(SLC_CTRL_RGN_OP_MASK, 0x1);
+
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_CTRL, ctrl);
+
+		/*
+		 * Lower bits are ignored, no need to clip
+		 * END needs to be setup before START (latter triggers the operation)
+		 * END can't be same as START, so add (l2_line_sz - 1) to sz
+		 */
+		end_addr = start_addr + size + CONFIG_SLC_LINE_SIZE - 1;
+
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_END, end_addr);
+		z_arc_v2_aux_reg_write(_ARC_V2_SLC_RGN_START, start_addr);
+
+		/* Make sure "busy" bit reports correct status, see STAR 9001165532 */
+		z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL);
+		while (z_arc_v2_aux_reg_read(_ARC_V2_SLC_CTRL) & SLC_CTRL_BS_MASK) {
+			/* Do Nothing */
+		}
+	}
+}
+
+__maybe_unused void slc_flush_and_invalidate_region(void *start_addr_ptr, size_t size)
```

Comment:
any reason for this function to not be 'static'? (why here the 'maybe unused', and not for the other ones?)